pywbem_mock/_methodprovider.py

```diff
@@ -19,129 +19,81 @@
-to override the InvokeMethod in :class:`~pywbem_mock.MethodProvider` with user
-methods defined in a subclass of :class:`~pywbem_mock.MethodProvider`.
+The following example implements method ``Method1`` defined in class
+``CIM_Foo_sub_sub`` that is defined as follows:
 
-This module contains the default MethodProvider class which provides the
-following:
+.. code-block:: text
 
-1. Definition of the provider type (``provider_type = "method"``).
-2. Definition of the API and return for the ``InvokeMethod`` method required in
-   the user-defined subclass.
+        [Description ("Subclass of CIM_Foo_sub")]
+    class CIM_Foo_sub_sub : CIM_Foo_sub {
 
-A user-defined method provider can be created as follows:
+        string cimfoo_sub_sub;
 
-1. Implement the subclass of :class:`~pywbem_mock.MethodProvider` with:
+            [Description("Sample method with input and output parameters")]
+        uint32 Method1(
+            [IN ( false), OUT, Description("Response param 2")]
+          string OutputParam2);
+    };
 
-   a. Constructor:
-      `__init__()` that takes as input at least the cimrepository
-      object parameter and passes it to the superclass and any other
-      constructor parameters the user-defined provider requires.  Since the
-      user creates the instance.
+The example implementation of method ``Method1`` in the user-defined method
+provider modifies the value of property ``cimfoo_sub_sub`` of the instance,
+and returns it as its output parameter ``OutputParam2``:
 
-      .. code-block:: python
+.. code-block:: python
 
-         def __init__(self, cimrepository):
-             super(UserMethodTestProvider, self).__init__(cimrepository)
+    from pywbem import CIMInstanceName, CIMError, \\
+        CIM_ERR_INVALID_PARAMETER, CIM_ERR_METHOD_NOT_AVAILABLE
+    from pywbem_mock import MethodProvider
 
-   b. Definition of the CIM class the provider supports as a class variable.
+    class MyMethodProvider(MethodProvider):
 
-      .. code-block:: python
+        provider_classname = 'CIM_Foo_sub_sub'
 
-        provider_classnames = 'CIM_Foo'
+        def InvokeMethod(self, namespace, MethodName, ObjectName, Params):
 
-   c. Definition and implementation of the WBEM operations supported by the
-      provider from the set of request operations defined in the table above
-      for a specific provider type. This must be a subset of the methods
-      defined for the provider type. Methods that do not exist in the
-      user-defined provider default to the default method. Each of these
-      methods may:
+            if MethodName.lower() == 'method1':
+                if not isinstance(ObjectName, CIMInstanceName):
+                    raise CIMError(
+                        CIM_ERR_INVALID_PARAMETER,
+                        "CIM method {0} must be invoked on a CIM instance".
+                        format(MethodName))
+                return self.Tick(namespace, ObjectName, Params)
```

Comment:
What is a self.Tick(.

Reply:
oops. That is a leftover from an earlier version where I had a counter property and a Tick() method that increments the counter. Until I came to the point that using an existing MOF is probably better, even though the example is more abstract.

Reply:
DONE - changed back to "Method1".